// reason-expo/src/Constants.bs.js
// Generated by BUCKLESCRIPT, PLEASE EDIT WITH CARE

import * as ExpoConstants from "expo-constants";

var appOwnership = ExpoConstants.default.appOwnership;

var debugMode = ExpoConstants.default.debugMode;

var deviceId = ExpoConstants.default.deviceId;

var deviceName = ExpoConstants.default.deviceName;

var deviceYearClass = ExpoConstants.default.deviceYearClass;

var experienceUrl = ExpoConstants.default.experienceUrl;

var expoRuntimeVersion = ExpoConstants.default.expoRuntimeVersion;

var expoVersion = ExpoConstants.default.expoVersion;

var getWebViewUserAgentAsync = ExpoConstants.default.getWebViewUserAgentAsync;

var installationId = ExpoConstants.default.installationId;

var isDetached = ExpoConstants.default.isDetached;

var isDevice = ExpoConstants.default.isDevice;

var isHeadless = ExpoConstants.default.isHeadless;

var linkingUri = ExpoConstants.default.linkingUri;

var linkingUrl = ExpoConstants.default.linkingUrl;

var manifest = ExpoConstants.default.manifest;

var nativeAppVersion = ExpoConstants.default.nativeAppVersion;

var nativeBuildVersion = ExpoConstants.default.nativeBuildVersion;

var platform = ExpoConstants.default.platform;

var sessionId = ExpoConstants.default.sessionId;

var statusBarHeight = ExpoConstants.default.statusBarHeight;

var supportedExpoSdks = ExpoConstants.default.supportedExpoSdks;

var systemFonts = ExpoConstants.default.systemFonts;

export {
  appOwnership ,
  debugMode ,
  deviceId ,
  deviceName ,
  deviceYearClass ,
  experienceUrl ,
  expoRuntimeVersion ,
  expoVersion ,
  getWebViewUserAgentAsync ,
  installationId ,
  isDetached ,
  isDevice ,
  isHeadless ,
  linkingUri ,
  linkingUrl ,
  manifest ,
  nativeAppVersion ,
  nativeBuildVersion ,
  platform ,
  sessionId ,
  statusBarHeight ,
  supportedExpoSdks ,
  systemFonts ,
  
}
/* appOwnership Not a pure module */
