// reason-expo/src/MediaLibrary.bs.js
// Generated by BUCKLESCRIPT, PLEASE EDIT WITH CARE

import * as $$Array from "bs-platform/lib/es6/array.js";
import * as ExpoMediaLibrary from "expo-media-library";

var MediaType = {};

var SortBy = {};

function encodeSource(src) {
  return src.VAL;
}

var AlbumOption = {
  encodeSource: encodeSource
};

function encodeSource$1(src) {
  return src.VAL;
}

var AssetOption = {
  encodeSource: encodeSource$1
};

function getAssetInfoAsync(asset) {
  return ExpoMediaLibrary.getAssetInfoAsync(asset.VAL);
}

function deleteAssetsAsync(assets) {
  return ExpoMediaLibrary.deleteAssetsAsync($$Array.map((function (a) {
                    return a.VAL;
                  }), assets));
}

function createAlbumAsync(albumName, asset, copyAsset) {
  return ExpoMediaLibrary.createAlbumAsync(albumName, asset.VAL, copyAsset);
}

function deleteAlbumsAsync(albums, deleteAssets) {
  return ExpoMediaLibrary.deleteAlbumsAsync($$Array.map((function (a) {
                    return a.VAL;
                  }), albums), deleteAssets);
}

function addAssetsToAlbumAsync(assets, albums, deleteAssets) {
  return ExpoMediaLibrary.addAssetsToAlbumAsync(assets, $$Array.map((function (a) {
                    return a.VAL;
                  }), albums), deleteAssets);
}

function removeAssetsFromAlbumAsync(assets, albums) {
  return ExpoMediaLibrary.removeAssetsFromAlbumAsync(assets, $$Array.map((function (a) {
                    return a.VAL;
                  }), albums));
}

export {
  MediaType ,
  SortBy ,
  AlbumOption ,
  AssetOption ,
  getAssetInfoAsync ,
  deleteAssetsAsync ,
  createAlbumAsync ,
  deleteAlbumsAsync ,
  addAssetsToAlbumAsync ,
  removeAssetsFromAlbumAsync ,
  
}
/* expo-media-library Not a pure module */
